content/articles/updating-spatial-database-via-flask-restful/index.md

```diff
@@ -0,0 +1,191 @@
+```py
+api.add_resource(CoordinateId, '/api/coordinate/<int:coordinate_id>')
+```
+We can only pass an integer to the URL because we specify our variable type as an integer. The coordinate_id will be passed from the URL to the get method, which will be used to fetch the coordinate from our database.
+
+### Update Coordinate
+After retrieving a coordinate using its ID we can update it by sending a put request from Postman with the updated fields.
+```py
+ def put(self, coordinate_id):
+        response = {
+            "status": 204,
+            "message": "AOI is not available"
+        }
+        aoi_details = AoiCoordinate.query.filter_by(id=coordinate_id).first()
+        if aoi_details:
+            file = request.files['coordinate']
+            aoi = request.form.get('aoi')
+            read_file = file.read()
+            file_json = json.loads(read_file)
+            aoi_coordinate = file_json["features"][0]['geometry']
+
+            aoi_details.coordinate = aoi_coordinate
+            aoi_details.aoi = aoi
+            db.session.commit()
+
+            response['status'] = 200
+            response['message'] = "aoi updated successfully"
+            return response, 200
+        return response, 204
+```
+Just like the last get method, the Put method also accepts the coordinate_id that was passed from the URL, and it will use the same endpoint.
+
+### Delete
+This method is used for deleting the selected coordinate by sending a delete request from Postman. We will add the method to the CoordinateId class so that it will receive the coordinate_id.
+
+```py
+
+    def delete(self, coordinate_id):
+        response = {
+            "status": 204,
+            "message": "AOI is not available"
+        }
+        aoi = AoiCoordinate.query.filter_by(id=coordinate_id)
+        if aoi:
+            db.session.delete(aoi)
+            db.session.commit()
+            response['status'] = 200
+            response['message'] = 'Aoi deleted successfully'
+            return response, 200
+```
+
+If the coordinate is deleted successfully it will return the updated response code and status 200, but if not the default response will be returned.
+
+### Conclusion
+We have created an API that can be used for creating, reading, updating and deleting records from a database. It can be deployed for mobile, web and desktop apps.
+
+### Further Reading
+- [How to create a Geoserver and Geodatabase](https://www.section.io/engineering-education/how-to-create-a-geoserver-and-geodatabase/)
+- [Flask Restful](https://flask-restful.readthedocs.io/en/latest)
+- [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)
```

Comment:
Format these further references in the form: `Title - Source`. 

For example:

```
HTTP response status codes - MDN Docs
```

To be more detailed, copy the title (as is) from the source website.

Reply:
It's done